dialogy/plugins/text/classification/xlmr.py

```diff
@@ -257,14 +303,58 @@ def train(self, training_data: pd.DataFrame) -> None:
         training_data.loc[:, const.LABELS] = encoder.transform(
             training_data[const.LABELS]
         )
-        # Add state as an additonal field to text
+
+        # Append state to text
         if self.use_state:
-            training_data[const.TEXT] += "<s> " + training_data["state"] + " </s>"
+            # logger.debug(training_data[const.STATE])
+            training_data[const.TEXT] += "<s> " + training_data[self.state_column] + " </s>"
+
+        # Append prompt to text
+        if self.use_prompt:
+            logger.debug("Adding prompts to input text")
+
+            if not self.train_using_all_prompts:
+                for i in tqdm(range(training_data.shape[0]), desc="progress bar:"):
+                    _lang = training_data.iloc[i][self.lang_column]
+                    _state = training_data.iloc[i][self.state_column]
+                    _prompt =  self.lookup_prompt(lang=_lang, state=_state)[0]
```

Comment:
Here, we are employing a strategy to choose a single prompt out of all available prompts. This should live inside a function and reused across `train` and `inference` methods.